src/main/java/duke/Task.java

```diff
@@ -0,0 +1,52 @@
+package duke;
+
+import java.time.LocalDateTime;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Create a task class.
+     * @param description
+     */
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public boolean isDone() {
+        return this.isDone;
+    }
+
+    public void mark() {
+        this.isDone = true;
+    }
+
+    public void unmark() {
+        this.isDone = false;
+    }
```

Comment:
I think if to preserve the immutability concept taught in CS2030, `mark()` and `unnmark()` should create a new Task instead of changing its variables.